### PR TITLE
github-status: support custom message and error status

### PR DESCRIPTION
**Pull request by jdsn**
Adding support for the error status, meaning that the test run itself could not be run (while failure means, the test could be run but it failed, so the PR is breaking it).
Also adding support for custom message to be shown in github. Primarily used to describe the error status.

### CODE REVIEW

File: scripts/github-status/github-status.rb [hunk 1]
```diff
@@ -15,6 +15,19 @@ def initialize(config = {})
   end
 
   def create_status(commit, result, description = '', target_url = '')
+    description =  @comment_prefix + case result
+      when 'success'
```
Comment (tboerger) [inline scripts/github-status/github-status.rb]: Such things are usually symbols and not strings :)

File: scripts/github-status/github-status.rb [hunk 2]
```diff
@@ -154,15 +159,11 @@ def show_forcerebuild_pull_requests
...
-    case options[:status]
-      when 'success'
-        ghc.create_success_status(options[:sha], options[:target_url])
-      when 'failure', 'error'
-        ghc.create_failure_status(options[:sha], options[:target_url])
-      when 'pending'
-        ghc.create_pending_status(options[:sha], options[:target_url])
-    end
+    msg = options[:message]
+    msg = msg.nil? || msg.empty? ? '':msg
```
Comment (tboerger) [inline scripts/github-status/github-status.rb]:
```
msg = msg.nil? || msg.empty? ? '' : msg
```

File: scripts/github-status/github-status.rb [hunk 3]
```diff
@@ -154,15 +159,11 @@ def show_forcerebuild_pull_requests
...
+    turl = options[:target_url]
+    turl = turl.nil? || turl.empty? ? '':turl
```
Comment (tboerger) [inline scripts/github-status/github-status.rb]:
```
turl = turl.nil? || turl.empty? ? '' : turl
```

File: scripts/github-status/github-status.rb (hunk 2 above)
Comment (tboerger) [inline scripts/github-status/github-status.rb]: But you can make it even much simpler...

```
msg = options[:message].to_s
```

File: scripts/github-status/github-status.rb (hunk 3 above)
Comment (tboerger) [inline scripts/github-status/github-status.rb]: Same as above...

```
url = options[:target_url].to_s
```

File: scripts/github-status/github-status.rb [hunk 4]
```diff
@@ -15,6 +15,19 @@ def initialize(config = {})
   end
 
   def create_status(commit, result, description = '', target_url = '')
+    description =  @comment_prefix + case result
```
Comment (tboerger) [inline scripts/github-status/github-status.rb]:
```
description =  @comment_prefix + case result.to_sym
```

Comment (tboerger) [inline scripts/github-status/github-status.rb]: And don't forget to transform the ```when``` values to symbols.

File: scripts/github-status/github-status.rb (hunk 1 above)
Comment (jdsn) [inline scripts/github-status/github-status.rb]: they come from the command line arguments as strings, should I translate them to symbols then?

File: scripts/github-status/github-status.rb (hunk 2 above)
Comment (jdsn) [inline scripts/github-status/github-status.rb]: hehe, thanks

File: scripts/github-status/github-status.rb (hunk 1 above)
Comment (tboerger) [inline scripts/github-status/github-status.rb]: Yes, just result.to_sym should be fine as mentioned below.

File: scripts/github-status/github-status.rb [hunk 5]
```diff
@@ -152,17 +157,11 @@ def show_forcerebuild_pull_requests
     raise if options[:pr].nil? || options[:pr].empty?
     puts ghc.get_pr_latest_sha(options[:pr])
   when 'set-status'
-    raise unless ['success', 'failure', 'error', 'pending'].include? options[:status]
+    raise unless [:success, :failure, :error, :pending].include? options[:status]
```
Comment (tboerger) [inline scripts/github-status/github-status.rb]: In general you should raise a message or an error class :)

File: scripts/github-status/github-status.rb [hunk 6]
```diff
@@ -152,17 +157,11 @@ def show_forcerebuild_pull_requests
...
-    raise if options[:target_url].nil? || options[:target_url].empty?
-    case options[:status]
-      when 'success'
-        ghc.create_success_status(options[:sha], options[:target_url])
-      when 'failure', 'error'
-        ghc.create_failure_status(options[:sha], options[:target_url])
-      when 'pending'
-        ghc.create_pending_status(options[:sha], options[:target_url])
-    end
+    msg = options[:message]
```
Comment (tboerger) [inline scripts/github-status/github-status.rb]: In that case you can skip this variable definition and put ```options[:message]``` and ```options[:target_url]``` directly as a method parameter.

**Comment by bmwiedemann** [comment]
:+1:

File: scripts/github-status/github-status.rb (hunk 6 above)
Comment (jdsn) [inline scripts/github-status/github-status.rb]: thanks, and another two lines gone :)

**Comment by tboerger** [comment]
:+1:

**Pull request merged**